### PR TITLE
Forward v1alpha1 to v1alpha2 in order to rollback api version due to…

### PR DESCRIPTION
**Pull request by diemtvu**
… complication with config version roll-up.

Issue: https://github.com/istio/istio/issues/4471

### CODE REVIEW

File: authentication/v1alpha1/policy.proto [hunk 1]
```diff
@@ -120,55 +121,30 @@ message Jwt {
...
 
 // OriginAuthenticationMethod defines authentication method/params for origin
 // authentication. Origin could be end-user, device, delegate service etc.
-// Method should have unique name so they can be referred later in credential
-// rules. Currently, only JWT is supported for origin authentication.
+// Currently, only JWT is supported for origin authentication.
 message OriginAuthenticationMethod {
   // Jwt params for the method.
-  Jwt jwt = 1;
+  Jwt jwt = 2;
```
Comment (xiaolanz) [inline authentication/v1alpha1/policy.proto]: why it's 2?

File: authentication/v1alpha1/policy.proto [hunk 2]
```diff
@@ -183,174 +159,162 @@ message CredentialRule {
...
+// Last but not least, the principal binding rule defines which identity (peer
+// or origin) should be used as principal. By default, it uses peer.
 //
 // Examples:
+//
 // Policy to enable mTLS for all services in namespace frod
 //
 // ```yaml
-// apiVersion: authentication.istio.io/v1alpha1
+// apiVersion: authentication.istio.io/v1alpha2
```
Comment (xiaolanz) [inline authentication/v1alpha1/policy.proto]: This is v1alpha1.
Same for all examples.

File: authentication/v1alpha1/policy.proto (hunk 1 above)
Comment (diemtvu) [reply authentication/v1alpha1/policy.proto]: Artifact while moving things around I guess.

File: authentication/v1alpha1/policy.proto (hunk 2 above)
Comment (diemtvu) [reply authentication/v1alpha1/policy.proto]: Done

**Comment by costinm** [comment]
+andy for cherry picking

**Pull request merged**